rest/src/main/java/org/jboss/pnc/rest/restmodel/ProductVersionRest.java

```diff
@@ -158,6 +158,21 @@ public ProductVersion toProductVersion(Product product) {
 
     public ProductVersion toProductVersion(ProductVersion productVersion) {
         productVersion.setVersion(version);
+
+        if (currentProductMilestoneId != null) {
+            List<ProductMilestone> productMilestones = nullableStreamOf(productVersion.getProductMilestones())
+                    .filter(productMilestone -> productMilestone.getId().equals(currentProductMilestoneId))
+                    .collect(Collectors.toList());
+
+            if (!productMilestones.isEmpty()) {
+                productVersion.setCurrentProductMilestone(productMilestones.get(0));
+            } else {
+                productVersion.setCurrentProductMilestone(null);
+            }
+        } else {
```

Comment:
I wanted to make sure that if a milestoneId is updated to null from a non-null value, or if it is set to a non-existing id it would be set to null